### PR TITLE
client, daemon: introduce helper for querying snapd API for the list of slot/plug connections

### PR DESCRIPTION
**Pull request by bboozzoo**
Part of `snap conections work` from #6079. The PR cannot land without #6333 and the spread test will currently break.

### CODE REVIEW

**Comment by bboozzoo** [comment]
Heh, the `system` alias again.

File: client/connections.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	// Gadget is set for connections that were enabled by the gadget snap
+	Gadget bool `json:"gadget"`
+}
+
+// Connections contains information about connections, as well as related plugs
+// and slots
+type Connections struct {
+	// Established is the list of connections that are currently present
+	Established []Connection `json:"established"`
+	// Undersired is a list of connections that are explicitly denied
```
Comment (pedronis) [inline client/connections.go]: s/explicitly/manually/ ?

File: client/connections.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	Undesired []Connection `json:"undesired"`
+	Plugs     []Plug       `json:"plugs"`
+	Slots     []Slot       `json:"slots"`
+}
+
+type ConnectionOptions struct {
+	Snap string
+	// Name of slot or plug
```
Comment (pedronis) [inline client/connections.go]: this comment seems wrong now

File: client/connections.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	Interface string
+	All       bool
```
Comment (pedronis) [inline client/connections.go]: All probably merits a comment describing what it does

**Pull request merged**